DependencyInjection/HttplugExtension.php

```diff
@@ -278,4 +280,53 @@ private function registerDebugPlugin(ContainerBuilder $container, $name)
 
         return $serviceIdDebugPlugin;
     }
+
+    /**
+     * Make sure we inject the debug plugin for clients found by auto discovery.
+     *
+     * @param ContainerBuilder $container
+     * @param array            $config
+     */
+    private function configureAutoDiscoveryClients(ContainerBuilder $container, array $config)
+    {
+        $httpClient = $config['discovery']['client'];
+        if ($httpClient === 'auto') {
+            $httpClient = $this->registerAutoDiscoverableClientWithDebugPlugin($container, 'client');
+        } elseif ($httpClient !== null) {
```

Comment:
in this case i would do `elseif ($httpClient) {` so that setting it to `false` or `''` works as well. none of those values would be valid service references.